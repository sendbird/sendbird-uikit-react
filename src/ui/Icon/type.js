export default {
  ADD: 'ADD',
  ARROW_LEFT: 'ARROW_LEFT',
  ATTACH: 'ATTACH',
  BAN: 'BAN',
  BROADCAST: 'BROADCAST',
  CAMERA: 'CAMERA',
  CHANNELS: 'CHANNELS',
  CHAT: 'CHAT',
  CHAT_FILLED: 'CHAT_FILLED',
  CHEVRON_DOWN: 'CHEVRON_DOWN',
  CHEVRON_RIGHT: 'CHEVRON_RIGHT',
  CLOSE: 'CLOSE',
  COLLAPSE: 'COLLAPSE',
  COPY: 'COPY',
  CREATE: 'CREATE',
  DELETE: 'DELETE',
  DISCONNECTED: 'DISCONNECTED',
  DOCUMENT: 'DOCUMENT',
  DONE: 'DONE',
  DONE_ALL: 'DONE_ALL',
  DOWNLOAD: 'DOWNLOAD',
  EDIT: 'EDIT',
  EMOJI_MORE: 'EMOJI_MORE',
  ERROR: 'ERROR',
  EXPAND: 'EXPAND',
  FILE_AUDIO: 'FILE_AUDIO',
  FILE_DOCUMENT: 'FILE_DOCUMENT',
  FREEZE: 'FREEZE',
  GIF: 'GIF',
  INFO: 'INFO',
  LEAVE: 'LEAVE',
  MEMBERS: 'MEMBERS',
  MESSAGE: 'MESSAGE',
  MODERATIONS: 'MODERATIONS',
  MORE: 'MORE',
  MUTE: 'MUTE',
  NOTIFICATIONS: 'NOTIFICATIONS',
  NOTIFICATIONS_OFF_FILLED: 'NOTIFICATIONS_OFF_FILLED',
  OPERATOR: 'OPERATOR',
  PHOTO: 'PHOTO',
  PLAY: 'PLAY',
  PLUS: 'PLUS',
  QUESTION: 'QUESTION',
  REFRESH: 'REFRESH',
  REPLY: 'REPLY',
  REMOVE: 'REMOVE',
  SEARCH: 'SEARCH',
  SEND: 'SEND',
  SETTINGS_FILLED: 'SETTINGS_FILLED',
  SPINNER: 'SPINNER',
  SUPERGROUP: 'SUPERGROUP',
  THUMBNAIL_NONE: 'THUMBNAIL_NONE',
  TOGGLE_OFF: 'TOGGLE_OFF',
  TOGGLE_ON: 'TOGGLE_ON',
  USER: 'USER',
};
