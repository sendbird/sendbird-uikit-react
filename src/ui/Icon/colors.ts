/* eslint-disable no-redeclare */
export const Colors = {
  DEFAULT: 'DEFAULT',
  PRIMARY: 'PRIMARY',
  PRIMARY_2: 'PRIMARY_2',
  SECONDARY: 'SECONDARY',
  CONTENT: 'CONTENT',
  CONTENT_INVERSE: 'CONTENT_INVERSE',
  CONTENT_INVERSE_5: 'CONTENT_INVERSE_5',
  WHITE: 'WHITE',
  GRAY: 'GRAY',
  THUMBNAIL_ICON: 'THUMBNAIL_ICON',
  SENT: 'SENT',
  READ: 'READ',
  ON_BACKGROUND_1: 'ON_BACKGROUND_1',
  ON_BACKGROUND_2: 'ON_BACKGROUND_2',
  ON_BACKGROUND_3: 'ON_BACKGROUND_3',
  ON_BACKGROUND_4: 'ON_BACKGROUND_4',
  BACKGROUND_3: 'BACKGROUND_3',
  ERROR: 'ERROR',
} as const;
export type Colors = typeof Colors[keyof typeof Colors];

export default { Colors };
