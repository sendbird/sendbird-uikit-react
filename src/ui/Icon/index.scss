@import '../../styles/variables';

.sendbird-icon {
  display: inline-block;
  &:focus {
    outline: none;
  }
}

.sendbird-icon-color--primary [class*='fill'] {
  @include themed() {
    fill: t(primary-3);
  }
}
.sendbird-icon-color--primary-2 [class*='fill'] {
  @include themed() {
    fill: t(primary-2);
  }
}

.sendbird-icon-color--primary [class*='stroke'] {
  @include themed() {
    stroke: t(primary-3);
    stroke-width: 2px;
    stroke-linejoin: round;
  }
}

.sendbird-icon-color--secondary [class*='fill'] {
  @include themed() {
    fill: t(secondary-3);
  }
}

.sendbird-icon-color--content [class*='fill'] {
  @include themed() {
    fill: t(on-content-1);
  }
}

.sendbird-icon-color--content-inverse [class*='fill'] {
  @include themed() {
    fill: t(inverse-on-content-1);
  }
}

.sendbird-icon-color--content-inverse-5 [class*='fill'] {
  @include themed() {
    fill: t(inverse-on-content-5);
  }
}

.sendbird-icon-color--white [class*='fill'] {
  fill: #ffffff;
}

.sendbird-icon-color--gray [class*='fill'] {
  fill: var(--sendbird-light-onlight-04);
}

.sendbird-icon-color--thumbnail-icon [class*='fill'] {
  fill: var(--sendbird-light-onlight-02);
}

.sendbird-icon-color--sent [class*='fill'] {
  @include themed() {
    fill: t(inverse-on-content-3);
  }
}

.sendbird-icon-color--read [class*='fill'] {
  @include themed() {
    fill: t(secondary--3-2);
  }
}

.sendbird-icon-color--on-background-1 [class*='fill'] {
  @include themed() {
    fill: t(on-bg-1);
  }
}
.sendbird-icon-color--on-background-2 [class*='fill'] {
  @include themed() {
    fill: t(on-bg-2);
  }
}
.sendbird-icon-color--on-background-3 [class*='fill'] {
  @include themed() {
    fill: t(on-bg-3);
  }
}
.sendbird-icon-color--on-background-4 [class*='fill'] {
  @include themed() {
    fill: t(on-bg-4);
  }
}

.sendbird-icon-color--background-3 [class*='fill'] {
  @include themed() {
    fill: t(bg-3);
  }
}

.sendbird-icon-color--error [class*='fill'] {
  @include themed() {
    fill: t(error--3-2);
  }
}
