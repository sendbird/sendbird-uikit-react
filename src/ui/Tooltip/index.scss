@import '../../styles/variables';

.sendbird-tooltip {
  position: relative;
  display: inline-flex;
  justify-content: center;
  max-width: 200px;
  border-radius: 8px;
  padding: 8px 16px;
  @include themed() {
    background-color: t(bg--7-3);
  }

  &::after {
    content: '';
    position: absolute;
    bottom: -4px;
    width: 8px;
    height: 8px;
    transform: rotate(45deg);
    @include themed() {
      background-color: t(bg--7-3);
    }
  }

  &__text {
    font-family: var(--sendbird-font-family-default);
    font-size: 12px;
    font-weight: bold;
    font-stretch: normal;
    font-style: normal;
    line-height: 1.33;
    letter-spacing: normal;
  }
}
