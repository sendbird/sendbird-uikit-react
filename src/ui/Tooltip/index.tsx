import React, { ReactElement } from 'react';

import './index.scss';

import Label, { LabelColors, LabelTypography } from '../Label';

export interface TooltipProps {
  className?: string | Array<string>;
  children?: string | ReactElement;
}
export default function Tooltip({
  className = '',
  children= '',
}: TooltipProps): ReactElement {
  return (
    <div
      className={[
        ...(Array.isArray(className) ? className : [className]),
        'sendbird-tooltip',
      ].join(' ')}
    >
      <Label
        className="sendbird-tooltip__text"
        type={LabelTypography.CAPTION_2}
        color={LabelColors.ONCONTENT_1}
      >
        {children}
      </Label>
    </div>
  );
}
