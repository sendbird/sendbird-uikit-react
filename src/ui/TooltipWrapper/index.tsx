import React, { useState, useRef, ReactElement } from 'react';

import './index.scss';

const SPACE_FROM_TRIGGER = 8;

export interface TooltipWrapperProps {
  className?: string | Array<string>;
  children: ReactElement;
  hoverTooltip: ReactElement;
}
export default function TooltipWrapper({
  className = '',
  children,
  hoverTooltip,
  // clickTooltip can be added later
}: TooltipWrapperProps): ReactElement {
  const [showHoverTooltip, setShowHoverTooltip] = useState(false);
  const childrenRef = useRef(null);

  return (
    <div
      className={[
        ...(Array.isArray(className) ? className : [className]),
        'sendbird-tooltip-wrapper',
      ].join(' ')}
      onMouseOver={() => { setShowHoverTooltip(true); }}
      onFocus={() => { setShowHoverTooltip(true); }}
      onMouseOut={() => { setShowHoverTooltip(false); }}
      onBlur={() => { setShowHoverTooltip(false); }}
    >
      <div
        className="sendbird-tooltip-wrapper__children"
        ref={childrenRef}
      >
        {children}
      </div>
      {
        showHoverTooltip && (
          <div
            className="sendbird-tooltip-wrapper__hover-tooltip"
            style={{ bottom: `calc(100% + ${SPACE_FROM_TRIGGER}px)` }}
          >
            <div className="sendbird-tooltip-wrapper__hover-tooltip__inner">
              <div
                className="sendbird-tooltip-wrapper__hover-tooltip__inner__tooltip-container"
                style={{ left: childrenRef.current && `calc(${childrenRef.current.offsetWidth / 2}px - 50%)` }}
              >
                {hoverTooltip}
              </div>
            </div>
          </div>
        )
      }
    </div>
  );
}
