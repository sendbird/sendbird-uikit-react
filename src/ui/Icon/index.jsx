import React from 'react';
import PropTypes from 'prop-types';

import './index.scss';
import { Types } from './type';
import { Colors } from './colors';
import {
  changeColorToClassName,
  changeTypeToIconClassName,
} from './utils';

import IconAdd from '../../svgs/icon-add.svg';
import IconArrowLeft from '../../svgs/icon-arrow-left.svg';
import IconAttach from '../../svgs/icon-attach.svg';
import IconAudioOnLined from '../../svgs/icon-audio-on-lined.svg';
import IconBan from '../../svgs/icon-ban.svg';
import IconBroadcast from '../../svgs/icon-broadcast.svg';
import IconCamera from '../../svgs/icon-camera.svg';
import IconChannels from '../../svgs/icon-channels.svg';
import IconChat from '../../svgs/icon-chat.svg';
import IconChatFilled from '../../svgs/icon-chat-filled.svg';
import IconChevronDown from '../../svgs/icon-chevron-down.svg';
import IconChevronRight from '../../svgs/icon-chevron-right.svg';
import IconClose from '../../svgs/icon-close.svg';
import IconCollapse from '../../svgs/icon-collapse.svg';
import IconCopy from '../../svgs/icon-copy.svg';
import IconCreate from '../../svgs/icon-create.svg';
import IconDelete from '../../svgs/icon-delete.svg';
import IconDisconnected from '../../svgs/icon-disconnected.svg';
import IconDocument from '../../svgs/icon-document.svg';
import IconDone from '../../svgs/icon-done.svg';
import IconDoneAll from '../../svgs/icon-done-all.svg';
import IconDownload from '../../svgs/icon-download.svg';
import IconEdit from '../../svgs/icon-edit.svg';
import IconEmojiMore from '../../svgs/icon-emoji-more.svg';
import IconError from '../../svgs/icon-error.svg';
import IconExpand from '../../svgs/icon-expand.svg';
import IconFileAudio from '../../svgs/icon-file-audio.svg';
import IconFileDocument from '../../svgs/icon-file-document.svg';
import IconFreeze from '../../svgs/icon-freeze.svg';
import IconGif from '../../svgs/icon-gif.svg';
import IconInfo from '../../svgs/icon-info.svg';
import IconLeave from '../../svgs/icon-leave.svg';
import IconMembers from '../../svgs/icon-members.svg';
import IconMessage from '../../svgs/icon-message.svg';
import IconModerations from '../../svgs/icon-moderations.svg';
import IconMore from '../../svgs/icon-more.svg';
import IconMute from '../../svgs/icon-mute.svg';
import IconNotifications from '../../svgs/icon-notifications.svg';
import IconNotificationsOffFilled from '../../svgs/icon-notifications-off-filled.svg';
import IconOperator from '../../svgs/icon-operator.svg';
import IconPhoto from '../../svgs/icon-photo.svg';
import IconPlay from '../../svgs/icon-play.svg';
import IconPlus from '../../svgs/icon-plus.svg';
import IconQuestion from '../../svgs/icon-question.svg';
import IconRefresh from '../../svgs/icon-refresh.svg';
import IconRemove from '../../svgs/icon-remove.svg';
import IconReply from '../../svgs/icon-reply-filled.svg';
import IconSearch from '../../svgs/icon-search.svg';
import IconSend from '../../svgs/icon-send.svg';
import IconSettingsFilled from '../../svgs/icon-settings-filled.svg';
import IconSpinner from '../../svgs/icon-spinner.svg';
import IconSupergroup from '../../svgs/icon-supergroup.svg';
import IconThread from '../../svgs/icon-thread.svg';
import IconThumbnailNone from '../../svgs/icon-thumbnail-none.svg';
import IconToggleOff from '../../svgs/icon-toggleoff.svg';
import IconToggleOn from '../../svgs/icon-toggleon.svg';
import IconUser from '../../svgs/icon-user.svg';

function changeTypeToIconComponent(type) {
  switch (type) {
    case Types.ADD: return <IconAdd />;
    case Types.ARROW_LEFT: return <IconArrowLeft />;
    case Types.ATTACH: return <IconAttach />;
    case Types.AUDIO_ON_LINED: return <IconAudioOnLined />;
    case Types.BAN: return <IconBan />;
    case Types.BROADCAST: return <IconBroadcast />;
    case Types.CAMERA: return <IconCamera />;
    case Types.CHANNELS: return <IconChannels />;
    case Types.CHAT: return <IconChat />;
    case Types.CHAT_FILLED: return <IconChatFilled />;
    case Types.CHEVRON_DOWN: return <IconChevronDown />;
    case Types.CHEVRON_RIGHT: return <IconChevronRight />;
    case Types.CLOSE: return <IconClose />;
    case Types.COLLAPSE: return <IconCollapse />;
    case Types.COPY: return <IconCopy />;
    case Types.CREATE: return <IconCreate />;
    case Types.DELETE: return <IconDelete />;
    case Types.DISCONNECTED: return <IconDisconnected />;
    case Types.DOCUMENT: return <IconDocument />;
    case Types.DONE: return <IconDone />;
    case Types.DONE_ALL: return <IconDoneAll />;
    case Types.DOWNLOAD: return <IconDownload />;
    case Types.EDIT: return <IconEdit />;
    case Types.EMOJI_MORE: return <IconEmojiMore />;
    case Types.ERROR: return <IconError />;
    case Types.EXPAND: return <IconExpand />;
    case Types.FILE_AUDIO: return <IconFileAudio />;
    case Types.FILE_DOCUMENT: return <IconFileDocument />;
    case Types.FREEZE: return <IconFreeze />;
    case Types.GIF: return <IconGif />;
    case Types.INFO: return <IconInfo />;
    case Types.LEAVE: return <IconLeave />;
    case Types.MEMBERS: return <IconMembers />;
    case Types.MESSAGE: return <IconMessage />;
    case Types.MODERATIONS: return <IconModerations />;
    case Types.MORE: return <IconMore />;
    case Types.MUTE: return <IconMute />;
    case Types.NOTIFICATIONS: return <IconNotifications />;
    case Types.NOTIFICATIONS_OFF_FILLED: return <IconNotificationsOffFilled />;
    case Types.OPERATOR: return <IconOperator />;
    case Types.PHOTO: return <IconPhoto />;
    case Types.PLAY: return <IconPlay />;
    case Types.PLUS: return <IconPlus />;
    case Types.QUESTION: return <IconQuestion />;
    case Types.REFRESH: return <IconRefresh />;
    case Types.REMOVE: return <IconRemove />;
    case Types.REPLY: return <IconReply />;
    case Types.SEARCH: return <IconSearch />;
    case Types.SEND: return <IconSend />;
    case Types.SETTINGS_FILLED: return <IconSettingsFilled />;
    case Types.SPINNER: return <IconSpinner />;
    case Types.SUPERGROUP: return <IconSupergroup />;
    case Types.THREAD: return <IconThread />;
    case Types.THUMBNAIL_NONE: return <IconThumbnailNone />;
    case Types.TOGGLE_OFF: return <IconToggleOff />;
    case Types.TOGGLE_ON: return <IconToggleOn />;
    case Types.USER: return <IconUser />;
    default: return 'icon'; // If you see this text 'icon' replace icon for it
  }
}

export default function Icon({
  className,
  type,
  fillColor,
  width,
  height,
  onClick,
  children,
}) {
  const iconStyle = {
    width: typeof width === 'string' ? width : `${width}px`,
    minWidth: typeof width === 'string' ? width : `${width}px`,
    height: typeof height === 'string' ? height : `${height}px`,
    minHeight: typeof height === 'string' ? height : `${height}px`,
  };
  return (
    <div
      className={[
        ...Array.isArray(className) ? className : [className],
        'sendbird-icon',
        changeTypeToIconClassName(type),
        changeColorToClassName(fillColor),
      ].join(' ')}
      role="button"
      onClick={onClick}
      onKeyDown={onClick}
      tabIndex="0"
      style={iconStyle}
    >
      {children || changeTypeToIconComponent(type)}
    </div>
  );
}

Icon.propTypes = {
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string),
  ]),
  /** Type: Use strings from below list */
  type: PropTypes.oneOfType([
    PropTypes.oneOf(Object.keys(Types)),
    PropTypes.string,
  ]).isRequired,
  /** Type: Use Colors from below list */
  fillColor: PropTypes.oneOf(Object.keys(Colors)),
  width: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  height: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  onClick: PropTypes.func,
  children: PropTypes.element,
};

Icon.defaultProps = {
  className: '',
  fillColor: Colors.DEFAULT,
  width: 26,
  height: 26,
  onClick: () => { },
  children: null,
};

export const IconTypes = Types;
export const IconColors = Colors;
