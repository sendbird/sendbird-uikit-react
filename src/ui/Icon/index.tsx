import React from 'react';

import './index.scss';
import { Types } from './type';
import { Colors } from './colors';
import {
  changeColorToClassName,
  changeTypeToIconClassName,
} from './utils';

import IconAdd from '../../svgs/icon-add.svg';
import IconArrowLeft from '../../svgs/icon-arrow-left.svg';
import IconAttach from '../../svgs/icon-attach.svg';
import IconAudioOnLined from '../../svgs/icon-audio-on-lined.svg';
import IconBan from '../../svgs/icon-ban.svg';
import IconBroadcast from '../../svgs/icon-broadcast.svg';
import IconCamera from '../../svgs/icon-camera.svg';
import IconChannels from '../../svgs/icon-channels.svg';
import IconChat from '../../svgs/icon-chat.svg';
import IconChatFilled from '../../svgs/icon-chat-filled.svg';
import IconChevronDown from '../../svgs/icon-chevron-down.svg';
import IconChevronRight from '../../svgs/icon-chevron-right.svg';
import IconClose from '../../svgs/icon-close.svg';
import IconCollapse from '../../svgs/icon-collapse.svg';
import IconCopy from '../../svgs/icon-copy.svg';
import IconCreate from '../../svgs/icon-create.svg';
import IconDelete from '../../svgs/icon-delete.svg';
import IconDisconnected from '../../svgs/icon-disconnected.svg';
import IconDocument from '../../svgs/icon-document.svg';
import IconDone from '../../svgs/icon-done.svg';
import IconDoneAll from '../../svgs/icon-done-all.svg';
import IconDownload from '../../svgs/icon-download.svg';
import IconEdit from '../../svgs/icon-edit.svg';
import IconEmojiMore from '../../svgs/icon-emoji-more.svg';
import IconError from '../../svgs/icon-error.svg';
import IconExpand from '../../svgs/icon-expand.svg';
import IconFileAudio from '../../svgs/icon-file-audio.svg';
import IconFileDocument from '../../svgs/icon-file-document.svg';
import IconFreeze from '../../svgs/icon-freeze.svg';
import IconGif from '../../svgs/icon-gif.svg';
import IconInfo from '../../svgs/icon-info.svg';
import IconLeave from '../../svgs/icon-leave.svg';
import IconMembers from '../../svgs/icon-members.svg';
import IconMessage from '../../svgs/icon-message.svg';
import IconModerations from '../../svgs/icon-moderations.svg';
import IconMore from '../../svgs/icon-more.svg';
import IconMute from '../../svgs/icon-mute.svg';
import IconNotifications from '../../svgs/icon-notifications.svg';
import IconNotificationsOffFilled from '../../svgs/icon-notifications-off-filled.svg';
import IconOperator from '../../svgs/icon-operator.svg';
import IconPhoto from '../../svgs/icon-photo.svg';
import IconPlay from '../../svgs/icon-play.svg';
import IconPlus from '../../svgs/icon-plus.svg';
import IconQuestion from '../../svgs/icon-question.svg';
import IconRefresh from '../../svgs/icon-refresh.svg';
import IconRemove from '../../svgs/icon-remove.svg';
import IconReply from '../../svgs/icon-reply-filled.svg';
import IconSearch from '../../svgs/icon-search.svg';
import IconSend from '../../svgs/icon-send.svg';
import IconSettingsFilled from '../../svgs/icon-settings-filled.svg';
import IconSlideLeft from '../../svgs/icon-slide-left.svg';
import IconSpinner from '../../svgs/icon-spinner.svg';
import IconSupergroup from '../../svgs/icon-supergroup.svg';
import IconThread from '../../svgs/icon-thread.svg';
import IconThumbnailNone from '../../svgs/icon-thumbnail-none.svg';
import IconToggleOff from '../../svgs/icon-toggleoff.svg';
import IconToggleOn from '../../svgs/icon-toggleon.svg';
import IconUser from '../../svgs/icon-user.svg';
import IconFeedbackLike from '../../svgs/icon-feedback-like.svg';
import IconFeedbackDislike from '../../svgs/icon-feedback-dislike.svg';

import { noop } from '../../utils/utils';

function changeTypeToIconComponent(type: Types) {
  switch (type) {
    case Types.ADD: return <IconAdd />;
    case Types.ARROW_LEFT: return <IconArrowLeft />;
    case Types.ATTACH: return <IconAttach />;
    case Types.AUDIO_ON_LINED: return <IconAudioOnLined />;
    case Types.BAN: return <IconBan />;
    case Types.BROADCAST: return <IconBroadcast />;
    case Types.CAMERA: return <IconCamera />;
    case Types.CHANNELS: return <IconChannels />;
    case Types.CHAT: return <IconChat />;
    case Types.CHAT_FILLED: return <IconChatFilled />;
    case Types.CHEVRON_DOWN: return <IconChevronDown />;
    case Types.CHEVRON_RIGHT: return <IconChevronRight />;
    case Types.CLOSE: return <IconClose />;
    case Types.COLLAPSE: return <IconCollapse />;
    case Types.COPY: return <IconCopy />;
    case Types.CREATE: return <IconCreate />;
    case Types.DELETE: return <IconDelete />;
    case Types.DISCONNECTED: return <IconDisconnected />;
    case Types.DOCUMENT: return <IconDocument />;
    case Types.DONE: return <IconDone />;
    case Types.DONE_ALL: return <IconDoneAll />;
    case Types.DOWNLOAD: return <IconDownload />;
    case Types.EDIT: return <IconEdit />;
    case Types.EMOJI_MORE: return <IconEmojiMore />;
    case Types.ERROR: return <IconError />;
    case Types.EXPAND: return <IconExpand />;
    case Types.FILE_AUDIO: return <IconFileAudio />;
    case Types.FILE_DOCUMENT: return <IconFileDocument />;
    case Types.FREEZE: return <IconFreeze />;
    case Types.GIF: return <IconGif />;
    case Types.INFO: return <IconInfo />;
    case Types.LEAVE: return <IconLeave />;
    case Types.MEMBERS: return <IconMembers />;
    case Types.MESSAGE: return <IconMessage />;
    case Types.MODERATIONS: return <IconModerations />;
    case Types.MORE: return <IconMore />;
    case Types.MUTE: return <IconMute />;
    case Types.NOTIFICATIONS: return <IconNotifications />;
    case Types.NOTIFICATIONS_OFF_FILLED: return <IconNotificationsOffFilled />;
    case Types.OPERATOR: return <IconOperator />;
    case Types.PHOTO: return <IconPhoto />;
    case Types.PLAY: return <IconPlay />;
    case Types.PLUS: return <IconPlus />;
    case Types.QUESTION: return <IconQuestion />;
    case Types.REFRESH: return <IconRefresh />;
    case Types.REMOVE: return <IconRemove />;
    case Types.REPLY: return <IconReply />;
    case Types.SEARCH: return <IconSearch />;
    case Types.SEND: return <IconSend />;
    case Types.SETTINGS_FILLED: return <IconSettingsFilled />;
    case Types.SLIDE_LEFT: return <IconSlideLeft />;
    case Types.SPINNER: return <IconSpinner />;
    case Types.SUPERGROUP: return <IconSupergroup />;
    case Types.THREAD: return <IconThread />;
    case Types.THUMBNAIL_NONE: return <IconThumbnailNone />;
    case Types.TOGGLE_OFF: return <IconToggleOff />;
    case Types.TOGGLE_ON: return <IconToggleOn />;
    case Types.USER: return <IconUser />;
    case Types.FEEDBACK_LIKE: return <IconFeedbackLike />;
    case Types.FEEDBACK_DISLIKE: return <IconFeedbackDislike />;
    default: return 'icon'; // If you see this text 'icon' replace icon for it
  }
}

type IconProps = {
  className?: string | string[];
  /** Type: Use strings from below list */
  type: Types;
  /** Type: Use Colors from below list */
  fillColor?: Colors;
  width?: string | number;
  height?: string | number;
  onClick?: React.MouseEventHandler<HTMLDivElement> & React.KeyboardEventHandler<HTMLDivElement>;
  children?: React.ReactNode;
};
export default function Icon({
  className = '',
  type,
  fillColor = Colors.DEFAULT,
  width = 26,
  height = 26,
  onClick = noop,
  children = null,
}: IconProps) {
  const iconStyle = {
    width: typeof width === 'string' ? width : `${width}px`,
    minWidth: typeof width === 'string' ? width : `${width}px`,
    height: typeof height === 'string' ? height : `${height}px`,
    minHeight: typeof height === 'string' ? height : `${height}px`,
  };
  return (
    <div
      className={[
        ...(Array.isArray(className) ? className : [className]),
        'sendbird-icon',
        changeTypeToIconClassName(type),
        changeColorToClassName(fillColor),
      ].join(' ')}
      role={'button'}
      onClick={onClick}
      onKeyDown={onClick}
      tabIndex={0}
      style={iconStyle}
    >
      {children || changeTypeToIconComponent(type)}
    </div>
  );
}

export const IconTypes = Types;
export const IconColors = Colors;
