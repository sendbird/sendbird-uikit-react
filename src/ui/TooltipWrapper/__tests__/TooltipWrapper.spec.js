import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import TooltipWrapper from "../index";

import Icon, { IconTypes, IconColors, changeTypeToIconClassName, changeColorToClassName } from '../../Icon';
import Tooltip from '../../Tooltip';

const iconClassName = 'icon-class-name';
const tooltipClassName = 'tooltip-class-name';
const tooltipText = 'sample-text';
const tooltipWrapperClassName = 'tooltip-wrapper-class-name';

const tooltipWrapperComponent = (
  <TooltipWrapper
    className={tooltipWrapperClassName}
    hoverTooltip={<Tooltip className={tooltipClassName}>{tooltipText}</Tooltip>}
  >
    <Icon className={iconClassName} fillColor={IconColors.PRIMARY} type={IconTypes.ADD} />
  </TooltipWrapper>
);

describe('ui/TooltipWrapper', () => {
  it('should exist after mouseover', function () {
    const { container } = render(tooltipWrapperComponent);

    expect(
      container.getElementsByClassName('sendbird-tooltip-wrapper')[0].children[0].className
    ).toBe('sendbird-tooltip-wrapper__children');
    expect(
      container.getElementsByClassName('sendbird-tooltip-wrapper')[0].children[0].children[0].className
    ).toContain(iconClassName);
    expect(
      container.querySelectorAll('.sendbird-tooltip-wrapper__hover-tooltip__inner__tooltip-container')
    ).toHaveLength(0);
    fireEvent.mouseOver(container.querySelector('.sendbird-tooltip-wrapper'));
    expect(
      container.querySelectorAll('.sendbird-tooltip-wrapper__hover-tooltip__inner__tooltip-container')
    ).toHaveLength(1);
  });

  it('should contain tooltip after mouseover', function () {
    const { container } = render(tooltipWrapperComponent);

    expect(
      container.getElementsByClassName(iconClassName)
    ).toHaveLength(1);
    expect(
      container.getElementsByClassName('sendbird-icon')[0].className
    ).toContain(changeTypeToIconClassName(IconTypes.ADD));
    expect(
      container.getElementsByClassName('sendbird-icon')[0].className
    ).toContain(changeColorToClassName(IconColors.PRIMARY));

    expect(
      container.getElementsByClassName(tooltipClassName)
    ).toHaveLength(0);
    fireEvent.mouseOver(container.querySelector(`.${tooltipWrapperClassName}`));
    expect(
      container.getElementsByClassName(tooltipClassName)
    ).toHaveLength(1);
  })

  it('should do a snapshot test of the TooltipWrapper DOM', function () {
    const { asFragment } = render(
      <TooltipWrapper
        hoverTooltip={<Tooltip>Test Text</Tooltip>}
      >
        <Icon type={IconTypes.ADD} fillColor={IconColors.PRIMARY} />
      </TooltipWrapper>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
