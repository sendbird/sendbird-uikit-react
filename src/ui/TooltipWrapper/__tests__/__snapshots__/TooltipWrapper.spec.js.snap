// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipWrapper should do a snapshot test of the TooltipWrapper DOM 1`] = `
<div
  className=" sendbird-tooltip-wrapper"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="sendbird-tooltip-wrapper__children"
  >
    <div
      className=" sendbird-icon sendbird-icon-add sendbird-icon-color--primary"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "height": "26px",
          "minHeight": "26px",
          "minWidth": "26px",
          "width": "26px",
        }
      }
      tabIndex="0"
    >
      <test-file-stub />
    </div>
  </div>
</div>
`;
